src/test/java/com/epam/service/impl/TrainerServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package com.epam.service.impl;
+
+import com.epam.config.AppConfig;
+import com.epam.model.Trainer;
+import com.epam.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class TrainerServiceImplTest {
+
+    private TrainerServiceImpl trainerService;
+
+    @BeforeEach
+    public void setUp() {
+        ApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+        this.trainerService = context.getBean(TrainerServiceImpl.class);
```

Comment:
Is it really needed? I mean, that launching the whole `ApplicationContext` before each test could lead to the long amount of time spent on the test startup. And if the app grows, you will see how much time your tests take. Try to use only necessary classes for test. You can initialize this `TrainerServiceImpl` by hands I guess. It is allowed - totally.